### PR TITLE
Start up v2

### PR DESCRIPTION
**Pull request by greglaw95**
Added the ability to enter typical basal insulin usage, then receive timed notifications based on this information to remind the user when to take insulin.
Added a placeholder main page to the application to allow navigation.
Added a take insulin screen to allow the user to see their current insulin recommendation and to record taking insulin. Currently it allows the user to record either basal or bolus insulin but only provides recommendations for basal.

### CODE REVIEW

File: DMS/app/src/main/java/com/example/kbb12/dms/Model/InsulinModel.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import com.example.kbb12.dms.model.basalInsulinModel.DuplicateDoseException;
+import com.example.kbb12.dms.model.basalInsulinModel.BasalInsulinDose;
+
+import java.util.List;
+
+/**
+ * Created by kbb12 on 24/02/2017.
+ */
+public interface InsulinModel {
+    public void saveDoses(List<BasalInsulinDose> basicDoses,String basalInsulinBrandName) throws DuplicateDoseException;
```
Comment (liddad) [inline DMS/app/src/main/java/com/example/kbb12/dms/Model/InsulinModel.java:12]: Does the insulin model only deal with basal insulin just now, or is this separated?

**Review by liddad — CHANGES_REQUESTED** [review]
Are MainActivity & Main2Activity used? If so, can they be renamed? 
Do we have separate Models for basal & short acting insulin? (This is fine I just want to know) And to what extent are we giving recommendations? Do we give them/ do we improve them? Do we give for basal and short-acting or only short-acting?

File: DMS/app/src/main/java/com/example/kbb12/dms/Model/InsulinModel.java (hunk 1 above)
Comment (greglaw95) [reply DMS/app/src/main/java/com/example/kbb12/dms/Model/InsulinModel.java:12]: It'll be the same underlying model it's an interface for the UserModel. It probably will be separated actually this is really just the interface between the screen that takes in the basal information on first set-up and the global model.

**Comment by greglaw95** [comment]
I can't find Main2Activity anywhere. If it is in there it's probably a mistake where did you see it? Also actually MainActivity is used it's the one that decides whether the initial start-up is needed and either shows a welcome screen or re-directs to the main menu.

We will have separate models for the basal and bolus insulin in fact we'll probably have 3 tables which essentially make up the different mathematical models for the basal insulin, insulin to carb ratio and insulin sensitivity factors throughout the day. More than that we'll actually have duplicates of each one to hold the original values and one to hold improvements we make to it.

In this pull request we only have the table for the basal insulin set-up but there is code that will allow a duplicate to be set-up using an almost identical contract very quickly. At the moment the "recommendations" are essentially just what the user tells us on first start up. A notification is fired at the times they said they take insulin. The recommendation in the take insulin page is based on the last dose of insulin they were supposed to take but haven't registered as having taken in the take insulin page yet.

There isn't any short-acting (I'm going to call it bolus from now on because technically it can be short-acting or rapid-acting) but I'm building that into the next pull request. I'm hoping to have the taking in and building the initial models done by early next week but for actual recommendations based on it though I'm going to need access to how we're storing blood glucose levels and some indication of how many carbs they've eaten.

I'll start work on improving the models using a hill-climber type technique once I have both bolus models up and running normally.

**Review by liddad — APPROVED** [review]

**Pull request merged**